src/main/java/org/example/persistence/PersistenceProgram.java

```diff
@@ -1,54 +1,422 @@
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return null;
+        }
+
+
     }
 
     @Override
-    public ProgramEntity getProgramById(String id) {
-        return null;
+    public ProgramEntity getProgramById(ProgramEntity programEntity) {
+        try {
+            PreparedStatement stmt = connection.prepareStatement("SELECT programinformation.description ," +
+                    " programinformation.title , programinformation.program_id FROM programinformation, program  where programinformation.program_id = ? ");
+            stmt.setLong(1, programEntity.getPrograminformationId());
+
+            var resultSet = stmt.executeQuery();
+
+
+            // checks if resultset contains any rows
+            if (!resultSet.next()) {
+                return null;
+            }
+            ProgramEntity returnValue = new ProgramEntity(resultSet.getLong("program_id"),resultSet.getString("title"),
+                    resultSet.getString("description"), getCompanyProgram(programEntity),
+                    getProducerforProgram(programEntity), getCreditUser(programEntity));
+
+
+            return returnValue;
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return null;
+        }
     }
 
     @Override
     public List<ProgramEntity> getAllPrograms() {
-        return null;
+        PreparedStatement stmt = null;
+        try {
+            stmt = connection.prepareStatement("SELECT programinformation.title , " +
+                    "programinformation.program_id from programinformation");
+
+            ResultSet sqlReturnValues = stmt.executeQuery();
+            List<ProgramEntity> returnvalue = new ArrayList<>();
+
+            while (sqlReturnValues.next()) {
+                
+                returnvalue.add(new ProgramEntity(sqlReturnValues.getLong("program_id"),sqlReturnValues.getString("title")));
+
+
+            }
+            return returnvalue;
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return null;
+        }
     }
 
     @Override
     public List<ProgramEntity> getProgramsByProducer(UserEntity producer) {
-        return null;
+
+        try {
+            PreparedStatement stmt = connection.prepareStatement("SELECT producer_id, programproducer.program_id, title from programproducer" +
+                    " inner join programinformation on programproducer.program_id = programinformation.program_id " +
+                    "where programproducer.producer_id = ?");
+            stmt.setLong(1, producer.getId());
+            ResultSet sqlReturn = stmt.executeQuery();
+            ArrayList<ProgramEntity> programList = new ArrayList<>();
+           while (sqlReturn.next()){
+               ProgramEntity program = new ProgramEntity(sqlReturn.getLong("program_id"), sqlReturn.getString("title"));
+               programList.add(program);
+
+           }
+
+            return programList;
```

Comment:
Hvorfor beder du om et helt object, når du kun bruger id'et for det? hvad er tanken bag det?